client/blocks/privacy-policy-banner/index.jsx

```diff
@@ -1,56 +1,70 @@
 /** @format */
 /**
  * External dependencies
- *
  */
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
 import { get, identity } from 'lodash';
 
 /**
  * Internal dependencies
  */
+import { requestHttpData } from 'state/data-layer/http-data';
+import { http } from 'state/data-layer/wpcom-http/actions';
 import { getPreference, isFetchingPreferences } from 'state/preferences/selectors';
 import { savePreference } from 'state/preferences/actions';
+import getCurrentUserRegisterDate from 'state/selectors/get-current-user-register-date';
 import Banner from 'components/banner';
 import config from 'config';
 import PrivacyPolicyDialog from './privacy-policy-dialog';
-import QueryPrivacyPolicy from 'components/data/query-privacy-policy';
 
-import getCurrentUserRegisterDate from 'state/selectors/get-current-user-register-date';
-import getPrivacyPolicyByEntity from 'state/selectors/get-privacy-policy-by-entity';
+const AUTOMATTIC_ENTITY = 'automattic';
+const PRIVACY_POLICY_PREFERENCE = 'privacy_policy';
+const PRIVACY_POLICY_REQUEST_ID = 'privacy-policy';
+
+const privacyPolicyFetchAction = http( {
+	method: 'GET',
+	path: '/privacy-policy',
+	apiNamespace: 'wpcom/v2',
+} );
 
-import { AUTOMATTIC_ENTITY, PRIVACY_POLICY_PREFERENCE } from './constants';
+// extract the "automattic" policy from the list of entities and ignore the other ones
+const privacyPolicyFromApi = () => data => [
+	[ PRIVACY_POLICY_REQUEST_ID, get( data, [ 'entities', AUTOMATTIC_ENTITY ], null ) ],
```

Comment:
is there any reason here to keep the constants or not inline the `http()` action? also in `fromApi` I don't see much reason for `get()` as any thrown exception will auto-fail-over to a failed network call

```js
requestHttpData(
	'privacy-policy',
	http( { method: 'GET', path: '/privacy-policy', apiNamespace: 'wpcom/v2' } ),
	{
		fromApi: () => data => [ [ 'privacy-policy', data.entities.automattic ] ]
	}
);
```

Reply:
> is there any reason here to keep the constants or not inline the http() action?

I want to avoid creating all these objects on every Redux state update. I think it's a design flaw of the `requestHttpData` API that there is no distinction between "specifying what should happen when fetching" and "performing a check with a potential fetch". We should eventually fix it and follow the example of `@wordpress/data` resolvers and React team's `simple-cache-provider` -- these are very similar API that don't have this problem.

> I don't see much reason for get() as any thrown exception will auto-fail-over to a failed network call

No no no, my code is not going to throw `ReferenceError`s in production. Sorry 😅 I was thinking about adding a schema validation for this REST endpoint, but then decided that it's too complicated and defeats the simplicity of `requestHttpData`.

Reply:
> No no no, my code is not going to throw ReferenceErrors in production. Sorry 😅 I was thinking about adding a schema validation for this REST endpoint, but then decided that it's too complicated and defeats the simplicity of requestHttpData.

I'm confused at what you are saying. Inside of `fromApi` we _want_ to throw errors on invalid API schema. It won't propagate past the `http-data` system. Instead, it will return an updated value with `state: failed` and an `error` property with the thrown value.

> API that there is no distinction between "specifying what should happen when fetching" and "performing a check with a potential fetch".

Can you explain more? The design goals here were specifically for addressing one-off data requests and fetching data that won't interact with the app other than being read from.

If the `http()` call and `fromApi` are already split out like you have them is there much going on on each update?